bin/get_config_env.py

```diff
@@ -1,19 +1,31 @@
-import requests,json, os
+import json
+import os
+
+import requests
 from requests.auth import HTTPBasicAuth
 
-base_host = os.environ.get('DISCOVERY_HOST') or u'gateway.watsonplatform.net'
 
-request_url = "https://{0}/discovery-experimental/api/v1".format(base_host)
+VCAP_SERVICES = json.loads(os.environ.get('VCAP_SERVICES'))
+VCAP_DISCOVERY_CREDENTIALS = VCAP_SERVICES['discovery'][0]['credentials']
+VCAP_DISCOVERY_URL = VCAP_DISCOVERY_CREDENTIALS['url']
+VCAP_DISCOVERY_USERNAME = VCAP_DISCOVERY_CREDENTIALS['username']
+VCAP_DISCOVERY_PASSWORD = VCAP_DISCOVERY_CREDENTIALS['password']
+
+base_host = os.environ.get('DISCOVERY_HOST', VCAP_DISCOVERY_URL)
+
+request_url = "https://{0}/discovery/api/v1".format(base_host)
+
 
 def get_auth():
-    uname = os.environ.get('DISCOVERY_USERNAME')
-    passwd = os.environ.get('DISCOVERY_PASSWORD')
-    return HTTPBasicAuth(uname,passwd)
+    uname = os.environ.get('DISCOVERY_USERNAME', VCAP_DISCOVERY_USERNAME)
+    passwd = os.environ.get('DISCOVERY_PASSWORD', VCAP_DISCOVERY_PASSWORD)
+    return HTTPBasicAuth(uname, passwd)
+
 
 def get_news_collection(env_id):
     try:
         response = requests.get(
-            url="{0}/environments/{1}/collections".format(request_url,env_id),
+            url="{0}/environments/{1}/collections".format(request_url, env_id),
             params={
                 "version": "2016-11-09",
```

Comment:
this version probably needs to be updated

Reply:
Am I supposed to change it to today's date or something? I have no idea how that parameter is used by the API.

Reply:
The latest API version is `2017-11-07` see here: https://www.ibm.com/watson/developercloud/discovery/api/v1/#authentication 

i have no idea why or how it's picked

Reply:
also, @rhagarty may be able to shed some light on that magic value